### PR TITLE
Option allowing to use simple lookupTransform API

### PR DESCRIPTION
**Pull request by AlexeyMerzlyakov**
ignoring time shifts between source and base frame during the movement
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #3402 |
| Primary OS tested on | Ubuntu 20.04 with ROS2 Rolling built from sources |
| Robotic platform tested on | TB3 Gazebo simulation, `colcon test --packages-select nav2_collision_monitor`, code coverage |

---

## Description of contribution in a few bullet points

* Added an new option allowing to control which `lookupTransform()` API to use: [simple](https://docs.ros2.org/latest/api/tf2/classtf2_1_1BufferCore.html#a5d578ecbea2ec0fdb92de1616251e175) or [advanced](https://docs.ros2.org/latest/api/tf2/classtf2_1_1BufferCore.html#a779c28b06b41f9ebc63e8c0d0da77e9b), considering considering the difference between current time and latest source time and allowing to correct source data towards to base frame movement.
* Using advanced `lookupTransform()` API will lead to better accuracy in cost of performance (strongly recommended for fast moving robots)
* Using simple `lookupTransform()` API will lower the accuracy but will improve the performance for `~1/(2*odom_rate)` per each `cmd_vel` calculation cycle, which might be suitable for slow robots.
* Using of simple `lookupTransform()` API is being covered by `testIgnoreTimeShift` testcase (advanced `lookupTransform()` API is already covered by previous tests)

## Description of documentation updates required from your changes

* Added new parameter on `Collision Monitor` configuration page

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by AlexeyMerzlyakov** [comment]
Local code coverage results:
Screenshot 2023-02-16 at 12-01-45 LCOV - cm info

File: nav2_collision_monitor/src/pointcloud.cpp [hunk 1]
```diff
@@ -76,16 +77,29 @@ void PointCloud::getData(
...
+      !nav2_util::getTransform(
+        data_->header.frame_id, data_->header.stamp,
+        base_frame_id_, curr_time, global_frame_id_,
+        transform_tolerance_, tf_buffer_, tf_transform))
+    {
+      return;
+    }
+  } else {
+    // Obtaining the transform to get data from source frame to base frame without time shift
+    // considered. Less accurate but much more faster option.
```
Comment (SteveMacenski) [inline nav2_collision_monitor/src/pointcloud.cpp:95]:
```suggestion
    // considered. Less accurate but much more faster option not dependent on state estimation frames.
```

Ditto in other locations - this is mostly a note for myself in the future

**Review by SteveMacenski — APPROVED** [review]
LGTM otherwise

File: nav2_collision_monitor/src/pointcloud.cpp (hunk 1 above)
Comment (AlexeyMerzlyakov) [reply nav2_collision_monitor/src/pointcloud.cpp]: Done in https://github.com/ros-planning/navigation2/pull/3412/commits/7e432ecf90f8982cf38f88a527406becf5e9090c

**Pull request merged**